providers/amazon/tests/unit/amazon/aws/bundles/test_s3.py

```diff
@@ -0,0 +1,293 @@
+from airflow.models import Connection
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.utils import db
+
+from tests_common.test_utils.config import conf_vars
+from tests_common.test_utils.db import clear_db_connections
+
+AWS_CONN_ID_WITH_REGION = "s3_dags_connection"
+AWS_CONN_ID_REGION = "eu-central-1"
+AWS_CONN_ID_DEFAULT = "aws_default"
+S3_BUCKET_NAME = "my-airflow-dags-bucket"
+S3_BUCKET_PREFIX = "project1/dags"
+
+if airflow.version.version.strip().startswith("3"):
+    from airflow.providers.amazon.aws.bundles.s3 import S3DagBundle
+
+
+@pytest.fixture
+def mocked_s3_resource():
+    with mock_aws():
+        yield boto3.resource("s3")
+
+
+@pytest.fixture
+def s3_client():
+    with mock_aws():
+        yield boto3.client("s3")
+
+
+@pytest.fixture
+def s3_bucket(mocked_s3_resource, s3_client):
+    bucket = mocked_s3_resource.create_bucket(Bucket=S3_BUCKET_NAME)
+
+    s3_client.put_object(Bucket=bucket.name, Key=S3_BUCKET_PREFIX + "/dag_01.py", Body=b"test data")
+    s3_client.put_object(Bucket=bucket.name, Key=S3_BUCKET_PREFIX + "/dag_02.py", Body=b"test data")
+    s3_client.put_object(
+        Bucket=bucket.name, Key=S3_BUCKET_PREFIX + "/subproject1/dag_a.py", Body=b"test data"
+    )
+    s3_client.put_object(
+        Bucket=bucket.name, Key=S3_BUCKET_PREFIX + "/subproject1/dag_b.py", Body=b"test data"
+    )
+
+    return bucket
+
+
+@pytest.fixture(autouse=True)
+def bundle_temp_dir(tmp_path):
+    with conf_vars({("dag_processor", "dag_bundle_storage_path"): str(tmp_path)}):
+        yield tmp_path
+
+
+@pytest.mark.skipif(not airflow.version.version.strip().startswith("3"), reason="Airflow >=3.0.0 test")
+class TestS3DagBundle:
+    @classmethod
+    def teardown_class(cls) -> None:
+        clear_db_connections()
+
+    @classmethod
+    def setup_class(cls) -> None:
+        db.merge_conn(
+            Connection(
+                conn_id=AWS_CONN_ID_DEFAULT,
+                conn_type="aws",
+                extra={
+                    "config_kwargs": {"s3": {"bucket_name": S3_BUCKET_NAME}},
+                },
+            )
+        )
+        db.merge_conn(
+            conn=Connection(
+                conn_id=AWS_CONN_ID_WITH_REGION,
+                conn_type="aws",
+                extra={
+                    "config_kwargs": {"s3": {"bucket_name": S3_BUCKET_NAME}},
+                    "region_name": AWS_CONN_ID_REGION,
+                },
+            )
+        )
+
+    @pytest.mark.db_test
+    def test_view_url_generates_presigned_url(self):
+        bundle = S3DagBundle(
+            name="test", aws_conn_id=AWS_CONN_ID_DEFAULT, prefix="project1/dags", bucket_name=S3_BUCKET_NAME
+        )
+        url: str = bundle.view_url("test_version")
+        assert url.startswith("https://my-airflow-dags-bucket.s3.amazonaws.com/project1/dags")
+        assert "AWSAccessKeyId=" in url
+        assert "Signature=" in url
+        assert "Expires=" in url
```

Comment:
Yeah, these are exactly what I would _not_ expect to see in the urls we give out to folks in the UI.

Reply:
Changed to normal http url